src/java.base/share/classes/sun/security/action/GetPropertyAction.java

```diff
@@ -160,4 +161,58 @@ public Properties run() {
             );
         }
     }
+
+    /**
+     * Convenience method for fetching System property values that are timeouts.
+     * Accepted timeout values may be purely numeric, a numeric value
+     * followed by "s" (both interpreted as seconds), or a numeric value
+     * followed by "ms" (interpreted as milliseconds).
+     *
+     * @param prop the name of the System property
+     * @param def a default value (in milliseconds)
+     * @param dbg a Debug object, if null no debug messages will be sent
+     *
+     * @return an integer value corresponding to the timeout value in the System
+     *      property in milliseconds.  If the property value is empty, negative,
+     *      or contains non-numeric characters (besides a trailing "s" or "ms")
+     *      then the default value will be returned.  If a negative value for
+     *      the "def" parameter is supplied, zero will be returned if the
+     *      property's value does not conform to the allowed syntax.
+     */
+    public static int privilegedGetTimeoutProp(String prop, int def, Debug dbg) {
+        if (def < 0) {
+            def = 0;
+        }
+
+        String propVal = privilegedGetProperty(prop, "").trim();
+        if (propVal.length() == 0) {
+            return def;
+        }
+
+        // Determine if "ms" or just "s" is on the end of the string
+        boolean isMillis = false;
+        if (propVal.toLowerCase().endsWith("ms")) {
+            propVal = propVal.substring(0, propVal.length() - 2);
+            isMillis = true;
+        } else if (propVal.toLowerCase().endsWith("s")) {
+            propVal = propVal.substring(0, propVal.length() - 1);
+        }
+
+        // Next check to make sure the string is built only from digits
+        if (propVal.matches("^\\d+$")) {
+            try {
+                int timeout = Integer.parseInt(propVal);
+                return isMillis ? timeout : timeout * 1000;
+            } catch (NumberFormatException nfe) {
+                if (dbg != null) {
+                    dbg.println("Warning: Unexpected " + nfe +
+                            " for timeout value " + propVal +
+                            ". Using default value of " + def + " msec.");
+                }
```

Comment:
This would also be useful debug for the else case on line 214-216 if the value is not an integer.

Reply:
Sounds good, I can add that.